File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var cnt=0;
var goodNodes = function(root) {
    cnt=0;
    fun(root,root.val);
    return cnt;
};
function fun(root,val){
    if(root===null){
        return;
    }
    if(root.val >= val){
        val = root.val;
        cnt++;
    }
    fun(root.left,val);
    fun(root.right,val);
}

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
class Solution {
public:
    int cnt = 0;
    int goodNodes(TreeNode* root) {
        fun(root,INT_MIN);
        return cnt;
    }
    void fun(TreeNode* root,int val){
        if(root==nullptr){
            return;
        }
        if(root->val >= val) {
            val = root->val;
            cnt++;
        }
        
        fun(root->left,val);
        fun(root->right,val);
    }
};